pkg/edit/navigation_test.go

```diff
@@ -63,23 +66,102 @@ func TestNavigation_WidthRatio(t *testing.T) {
 	evals(f.Evaler, `@edit:navigation:width-ratio = 1 1 1`)
 	f.TTYCtrl.Inject(term.K('N', ui.Ctrl))
 	f.TestTTY(t,
-		"~"+string(os.PathSeparator)+"d> ", term.DotHere, "\n",
+		filepath.Join("~", "d"), "> ", term.DotHere, "\n",
 		" NAVIGATING  \n", Styles,
 		"************ ",
-		" d                a               ", Styles,
+		" d                a               \n", Styles,
 		"################ ++++++++++++++++ ",
+		"                  e              ", Styles,
+		"                 ////////////////",
 	)
 }
 
+// Test corner case: Inserting a selection when the CLI cursor is not at the
+// start of the edit buffer, but the preceding char is a space, does not
+// insert another space.
+func TestNavigationCornerCaseRedundantSpace(t *testing.T) {
```

Comment:
I'd prefer to leave out the "corner case" moniker when describing a unit test - too many unit tests would satisfy some definition of "corner case"; instead, describe completely what the behavior is being tested. In this case, I'd name this `TestNavigation_EnterDoesNotInsertSpaceAfterSpace`.